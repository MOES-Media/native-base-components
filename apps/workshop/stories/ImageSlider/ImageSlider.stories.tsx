import ImageSlider from '@moes-media/native-base-components-image-slider';
import { ComponentMeta, ComponentStory } from '@storybook/react-native';

const images = [
  {
    src: 'https://images.unsplash.com/photo-1538655776941-6dcf10909d18?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80',
  },
  {
    src: 'https://plus.unsplash.com/premium_photo-1680246615216-4b958369b5b1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
  },
  {
    src: 'https://images.unsplash.com/photo-1511818966892-d7d671e672a2?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1471&q=80',
  },
  {
    src: 'https://images.unsplash.com/photo-1547930275-405fd0fcc8f8?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
  },
  {
    src: 'https://images.unsplash.com/photo-1453227588063-bb302b62f50b?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
  },
];

const imagesWithCaptions = [
  {
    src: 'https://images.unsplash.com/photo-1538655776941-6dcf10909d18?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80',
    caption: 'Image 1',
  },
  {
    src: 'https://plus.unsplash.com/premium_photo-1680246615216-4b958369b5b1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
    caption: 'Image 2',
  },
  {
    src: 'https://images.unsplash.com/photo-1511818966892-d7d671e672a2?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1471&q=80',
    caption: 'Image 3',
  },
  {
    src: 'https://images.unsplash.com/photo-1547930275-405fd0fcc8f8?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
    caption: 'Image 4',
  },
  {
    src: 'https://images.unsplash.com/photo-1453227588063-bb302b62f50b?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
    caption: 'Image 5',
  },
];

const ImageSliderMeta: ComponentMeta<typeof ImageSlider> = {
  title: 'ImageSlider',
  component: ImageSlider,
  args: {
    data: images,
    withPagination: false,
    autoPlay: false,
  },
};

export default ImageSliderMeta;

type ImageSliderStory = ComponentStory<typeof ImageSlider>;

export const Default: ImageSliderStory = (args) => <ImageSlider {...args} />;
export const withPagination: ImageSliderStory = (args) => <ImageSlider {...args} withPagination />;
export const withCustomWidth: ImageSliderStory = (args) => <ImageSlider {...args} withPagination width={275}/>;
export const WithCaptions: ImageSliderStory = (args) => (
  <ImageSlider {...args} data={imagesWithCaptions} />
);
