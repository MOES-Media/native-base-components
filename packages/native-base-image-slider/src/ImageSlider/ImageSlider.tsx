import { Box, FlatList } from 'native-base';
import { FC, useEffect, useRef, useState } from 'react';
import { Dimensions, LayoutAnimation, FlatList as NativeFlatList } from 'react-native';
import { Pagination, SliderContentItem } from '@moes-media/native-base-components-atoms';
import { isMobile } from '@moes-media/native-base-components-utils';

const renderSeperator = (separatorWidth?: number) => () => <Box w={separatorWidth || 0} />;

const getDefaultWidth = () => (isMobile() ? Math.round(Dimensions.get('window').width) : 290);

interface ImageProps {
  src: string;
  caption?: string;
}

interface ImageSliderProps {
  autoPlay?: boolean;
  autoPlayInterval?: number;
  data: ImageProps[];
  separatorWidth?: number;
  withPagination?: boolean;
  width?: number;
}

const ImageSlider: FC<ImageSliderProps> = ({
  data,
  separatorWidth,
  withPagination,
  width,
  autoPlay,
  autoPlayInterval,
}) => {
  const imageSliderRef = useRef<NativeFlatList>(null);

  const [activeIdx, setActiveIdx] = useState(0);

  const sliderWidth = width || getDefaultWidth();
  const totalWidth = sliderWidth + (separatorWidth || 0);

  useEffect(() => {
    if (!autoPlay) {
      return () => ({});
    }
    const timerId = setInterval(() => {
      setActiveIdx(idx => {
        if (idx + 1 === data.length) {
          return 0;
        }
        return idx + 1;
      });
    }, autoPlayInterval || 3000);
    return () => {
      clearInterval(timerId);
    };
  }, [autoPlay, autoPlayInterval, data.length]);

  useEffect(() => {
    if (!imageSliderRef.current) return;
    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
    imageSliderRef.current.scrollToIndex({
      index: activeIdx,
      animated: true,
    });
  }, [activeIdx, imageSliderRef]);

  return (
    <Box w={totalWidth}>
      <FlatList
        ref={imageSliderRef}
        horizontal
        w={sliderWidth}
        pagingEnabled
        data={data}
        snapToInterval={totalWidth}
        renderItem={({ item, index: idx }) => (
          <SliderContentItem
            imageSource={item.src}
            key={`image-slider-content-${idx}`}
            width={sliderWidth}
            active={idx === activeIdx}
            caption={item.caption}
          />
        )}
        decelerationRate="fast"
        ItemSeparatorComponent={renderSeperator(separatorWidth)}
        keyExtractor={(_, idx) => `image-${idx}`}
        maxToRenderPerBatch={process.env.NODE_ENV === 'test' ? data.length : 1}
        initialScrollIndex={0}
        initialNumToRender={process.env.NODE_ENV === 'test' ? data.length : 1}
        showsHorizontalScrollIndicator={false}
        bounces={false}
        viewabilityConfig={{ viewAreaCoveragePercentThreshold: 50 }}
        getItemLayout={(_, index) => ({
          length: totalWidth,
          offset: totalWidth * index,
          index,
        })}
      />
      {withPagination && (
        <Box mt={isMobile() ? 18 : 3}>
          <Pagination
            numberOfItems={data.length}
            activeIndex={activeIdx}
            onIndicatorChange={setActiveIdx}
          />
        </Box>
      )}
    </Box>
  );
};

export default ImageSlider;
