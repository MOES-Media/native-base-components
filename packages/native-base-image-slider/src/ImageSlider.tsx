import { Box, FlatList } from 'native-base';
import { FC, useCallback, useEffect, useRef, useState } from 'react';
import { ImageSliderProps } from './types/ImageSlider';
import { Pagination, SliderContentItem } from './atoms';
import { isMobile } from './utils';
import { Dimensions, LayoutAnimation } from 'react-native';

const renderSeperator = (separatorWidth?: number) => () => <Box w={separatorWidth || 0} />;

const getDefaultWidth = () => (isMobile() ? Math.round(Dimensions.get('window').width) : 290);

const ImageSlider: FC<ImageSliderProps> = ({
  data,
  separatorWidth,
  withPagination,
  width,
  autoPlay,
  autoPlayInterval,
}) => {
  const imageSliderRef = useRef<any>(null);

  const [activeIdx, setActiveIdx] = useState(0);
  const [isAnimating, setIsAnimating] = useState(false);

  const sliderWidth = width ? width : getDefaultWidth();
  const totalWidth = sliderWidth + (separatorWidth ? separatorWidth : 0);

  useEffect(() => {
    if (!autoPlay) {
      return;
    }
    const timerId = setInterval(
      () => {
        setActiveIdx((idx) => {
          if (idx + 1 === data.length) {
            return 0;
          }
          return idx + 1;
        });
      },
      autoPlayInterval ? autoPlayInterval : 3000
    );
    return () => clearInterval(timerId);
  }, [autoPlay, autoPlayInterval]);

  useEffect(() => {
    if (!imageSliderRef.current) return;
    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
    imageSliderRef.current.scrollToIndex({
      index: activeIdx,
    });
    setTimeout(() => setIsAnimating(false), 250);
  }, [activeIdx, imageSliderRef]);

  return (
    <Box w={sliderWidth}>
      <FlatList
        ref={imageSliderRef}
        horizontal
        w={sliderWidth}
        pagingEnabled
        data={data}
        snapToInterval={totalWidth}
        renderItem={({ item, index: idx }) => (
          <SliderContentItem
            imageSource={item.src}
            key={`image-slider-content-${idx}`}
            width={sliderWidth}
            active={idx === activeIdx}
          />
        )}
        decelerationRate="fast"
        ItemSeparatorComponent={renderSeperator(separatorWidth)}
        keyExtractor={(_, idx) => `image-${idx}`}
        maxToRenderPerBatch={1}
        initialScrollIndex={0}
        initialNumToRender={1}
        showsHorizontalScrollIndicator={false}
        bounces={false}
        viewabilityConfig={{ viewAreaCoveragePercentThreshold: 50 }}
        getItemLayout={(_, index) => ({
          length: totalWidth,
          offset: totalWidth * index,
          index,
        })}
      />
      {withPagination && (
        <Box mt={18}>
          <Pagination
            numberOfItems={data.length}
            activeIndex={activeIdx}
            onIndicatorChange={setActiveIdx}
          />
        </Box>
      )}
    </Box>
  );
};

export default ImageSlider;
