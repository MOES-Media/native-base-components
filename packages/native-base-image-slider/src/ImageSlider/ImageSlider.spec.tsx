import { NativeBaseProvider, Text } from 'native-base';
import * as renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { isMobile } from '@moes-media/native-base-components-utils';
import ImageSlider from './index';

const images = [
  {
    src: 'https://images.unsplash.com/photo-1538655776941-6dcf10909d18?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80',
  },
  {
    src: 'https://plus.unsplash.com/premium_photo-1680246615216-4b958369b5b1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
  },
  {
    src: 'https://images.unsplash.com/photo-1511818966892-d7d671e672a2?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1471&q=80',
  },
  {
    src: 'https://images.unsplash.com/photo-1547930275-405fd0fcc8f8?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
  },
  {
    src: 'https://images.unsplash.com/photo-1453227588063-bb302b62f50b?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
  },
];

const imagesWithCaptions = [
  {
    src: 'https://images.unsplash.com/photo-1538655776941-6dcf10909d18?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80',
    caption: 'Image 1',
  },
  {
    src: 'https://plus.unsplash.com/premium_photo-1680246615216-4b958369b5b1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
    caption: 'Image 2',
  },
  {
    src: 'https://images.unsplash.com/photo-1511818966892-d7d671e672a2?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1471&q=80',
    caption: 'Image 3',
  },
  {
    src: 'https://images.unsplash.com/photo-1547930275-405fd0fcc8f8?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
    caption: 'Image 4',
  },
  {
    src: 'https://images.unsplash.com/photo-1453227588063-bb302b62f50b?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80',
    caption: 'Image 5',
  },
];

const inset = {
  frame: {
    width: 320,
    height: 640,
    x: 0,
    y: 0,
  },
  insets: {
    left: 0,
    right: 0,
    bottom: 0,
    top: 0,
  },
};

const testCases = [];

describe('ImageSlider: Mobile', () => {
  beforeAll(() => {
    // @ts-ignore
    isMobile.mockImplementation(() => true);
  });
  afterAll(() => jest.resetAllMocks());

  it('given a minimum amount of configuration, it should render correctly', () => {
    const { getByTestId } = render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <ImageSlider data={images} />
      </NativeBaseProvider>
    );
    expect(
      renderer
        .create(
          <NativeBaseProvider initialWindowMetrics={inset}>
            <ImageSlider data={images} />
          </NativeBaseProvider>
        )
        .toJSON()
    ).toMatchSnapshot();
    expect(getByTestId('mock-slider-content-item')).toBeTruthy();
    expect(() => getByTestId('mock-pagination')).toThrow();
  });

  it('given a configuration withPagination enabled, it should render the pagination correctly', () => {
    const { getByTestId } = render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <ImageSlider data={images} withPagination />
      </NativeBaseProvider>
    );

    expect(getByTestId('mock-slider-content-item')).toBeTruthy();
    expect(getByTestId('mock-pagination')).toBeTruthy();
    expect(
      renderer
        .create(
          <NativeBaseProvider initialWindowMetrics={inset}>
            <ImageSlider data={images} withPagination />
          </NativeBaseProvider>
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
