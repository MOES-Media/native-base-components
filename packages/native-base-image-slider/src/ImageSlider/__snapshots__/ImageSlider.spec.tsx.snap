// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSlider: Mobile given a configuration withPagination enabled, it should render the pagination correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "width": 750,
      }
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      bounces={false}
      contentContainerStyle={
        [
          {},
          {
            "dataSet": {},
          },
          {},
        ]
      }
      data={
        [
          {
            "src": "https://images.unsplash.com/photo-1538655776941-6dcf10909d18?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80",
          },
          {
            "src": "https://plus.unsplash.com/premium_photo-1680246615216-4b958369b5b1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80",
          },
          {
            "src": "https://images.unsplash.com/photo-1511818966892-d7d671e672a2?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1471&q=80",
          },
          {
            "src": "https://images.unsplash.com/photo-1547930275-405fd0fcc8f8?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80",
          },
          {
            "src": "https://images.unsplash.com/photo-1453227588063-bb302b62f50b?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80",
          },
        ]
      }
      dataSet={{}}
      decelerationRate="fast"
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={true}
      initialNumToRender={1}
      initialScrollIndex={0}
      keyExtractor={[Function]}
      maxToRenderPerBatch={1}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      snapToInterval={750}
      stickyHeaderIndices={[]}
      style={
        {
          "width": 750,
        }
      }
      viewabilityConfig={
        {
          "viewAreaCoveragePercentThreshold": 50,
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <mock-slider-content-item
            active={true}
            data-testid="mock-slider-content-item"
            imageSource="https://images.unsplash.com/photo-1538655776941-6dcf10909d18?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80"
            width={750}
          />
          <View
            dataSet={{}}
            style={
              {
                "width": 0,
              }
            }
          />
        </View>
        <View
          style={
            {
              "width": 3000,
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      dataSet={{}}
      style={
        {
          "marginTop": 18,
        }
      }
    >
      <mock-pagination
        activeIndex={0}
        data-testid="mock-pagination"
        numberOfItems={5}
        onIndicatorChange={[Function]}
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`ImageSlider: Mobile given a minimum amount of configuration, it should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "width": 750,
      }
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      bounces={false}
      contentContainerStyle={
        [
          {},
          {
            "dataSet": {},
          },
          {},
        ]
      }
      data={
        [
          {
            "src": "https://images.unsplash.com/photo-1538655776941-6dcf10909d18?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80",
          },
          {
            "src": "https://plus.unsplash.com/premium_photo-1680246615216-4b958369b5b1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80",
          },
          {
            "src": "https://images.unsplash.com/photo-1511818966892-d7d671e672a2?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1471&q=80",
          },
          {
            "src": "https://images.unsplash.com/photo-1547930275-405fd0fcc8f8?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80",
          },
          {
            "src": "https://images.unsplash.com/photo-1453227588063-bb302b62f50b?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80",
          },
        ]
      }
      dataSet={{}}
      decelerationRate="fast"
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={true}
      initialNumToRender={1}
      initialScrollIndex={0}
      keyExtractor={[Function]}
      maxToRenderPerBatch={1}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      snapToInterval={750}
      stickyHeaderIndices={[]}
      style={
        {
          "width": 750,
        }
      }
      viewabilityConfig={
        {
          "viewAreaCoveragePercentThreshold": 50,
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <mock-slider-content-item
            active={true}
            data-testid="mock-slider-content-item"
            imageSource="https://images.unsplash.com/photo-1538655776941-6dcf10909d18?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=687&q=80"
            width={750}
          />
          <View
            dataSet={{}}
            style={
              {
                "width": 0,
              }
            }
          />
        </View>
        <View
          style={
            {
              "width": 3000,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
